### PR TITLE
Application APIs

### PR DESCRIPTION
**Pull request by gsanchietti**
API:
- [x] get-module-info
- [x] get-configuration
- [x] get-status
  - [x] validate_output.json
- [x] update design documentation

 
See https://trello.com/c/qGGQGTjd/96-application-api-actions

### CODE REVIEW

File: core/imageroot/var/lib/nethserver/cluster/actions/get-module-info/validate-input.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  "$id": "http://schema.nethserver.org/cluster/info-module-input.json",
+  "description": "Retrieve the metadata information af a module",
+  "examples": [
+    {
+      "id": "dokuwiki"
+    }
+  ],
+  "type": "object",
+  "properties": {
+    "if": {
```
Comment (andre8244) [inline core/imageroot/var/lib/nethserver/cluster/actions/get-module-info/validate-input.json:13]:
```suggestion
    "id": {
```

File: doc/details.md [hunk 2]
```diff
@@ -275,7 +275,15 @@ The following actions are automatically available for all modules:
 Each module can add extra step to the above actions by creating a directory with the of the action itself inside the module implementation,
 e.g. `imageroot/actions/myapp/create-module/20mystep`.
 
-If a module needs user input to have a valid default configuration, it should declare a `configure-module` action.
+##### Recommended actions
+
+As a convention, all modules with a UI should also implement the following actions:
+
+- `configure-module`: it validates the user input and applies the configuration, see [Dokuwiki example](../dokuwiki/imageroot/actions/configure-module)
+- `get-configuration`: it should return current configuration, the output should be equal or similar to configure-moduleinput, see [Dokuwiki example](../dokuwiki/imageroot/actions/get-configuration)
```
Comment (andre8244) [inline doc/details.md:283]:
```suggestion
- `get-configuration`: it should return current configuration, the output should be equal or similar to configure-module input, see [Dokuwiki example](../dokuwiki/imageroot/actions/get-configuration)
```

File: dokuwiki/imageroot/actions/get-status/20read [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+try:
+    images = subprocess.run(["podman", "image", "list", "--format", "{{index .Names 0}},{{.Size}},{{.CreatedAt}}"], capture_output=True, text=True, check=True)
+    for i in images.stdout.split("\n"):
+        # skip empty lines
+        if not i:
+            continue
+        info = i.split(",")
+        status["images"].append({"name": info[0], "size": info[1], "created": info[2]})
+except:
+    pass
```
Comment (andre8244) [inline dokuwiki/imageroot/actions/get-status/20read:56]: This is ok if we want to ignore errors on podman images retrieval

File: dokuwiki/imageroot/actions/get-status/20read [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+try:
+    images = subprocess.run(["podman", "volume", "ls", "--format", "{{.Name}},{{.Mountpoint}},{{.CreatedAt}}"], capture_output=True, text=True, check=True)
+    for i in images.stdout.split("\n"):
+        # skip empty lines
+        if not i:
+            continue
+        info = i.split(",")
+        status["volumes"].append({"name": info[0], 'mount': info[1], "created": info[2]})
+except:
+    pass
```
Comment (andre8244) [inline dokuwiki/imageroot/actions/get-status/20read:68]: Same as previous comment

**Review by andre8244 — CHANGES_REQUESTED** [review]

File: dokuwiki/imageroot/actions/get-status/20read (hunk 3 above)
Comment (gsanchietti) [reply dokuwiki/imageroot/actions/get-status/20read:56]: Yes, that's the idea. In case of error the JSON will be still valid

**Review by DavidePrincipi — APPROVED** [review]
`get-status` could be generalized and moved as a basic agent action and inherited by all modules.

OK to merge it now, and improve it in a later step.

File: core/imageroot/usr/local/nethserver/agent/actions/get-status/validate-input.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "get-status input",
+    "$id": "http://schema.nethserver.org/module/get-status-input.json",
```
Comment (DavidePrincipi) [inline core/imageroot/usr/local/nethserver/agent/actions/get-status/validate-input.json:4]: URL space for inherited actions:

```suggestion
    "$id": "http://schema.nethserver.org/agent/get-status-input.json",
```

File: core/imageroot/usr/local/nethserver/agent/actions/get-status/validate-output.json [hunk 6]
```diff
@@ -0,0 +1,119 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Get module status",
+  "$id": "http://nethserver.org/json-schema/task/input/module/get-status",
```
Comment (DavidePrincipi) [inline core/imageroot/usr/local/nethserver/agent/actions/get-status/validate-output.json:4]:
```suggestion
  "$id": ""http://schema.nethserver.org/agent/get-status-output.json",
```

File: dokuwiki/imageroot/actions/get-status/20read [hunk 7]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
```
Comment (DavidePrincipi) [inline dokuwiki/imageroot/actions/get-status/20read:1]: This override should be removed?

File: dokuwiki/imageroot/actions/get-status/validate-output.json [hunk 8]
```diff
@@ -0,0 +1,119 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Get Dokuwiki status",
+  "$id": "http://nethserver.org/json-schema/task/input/dokuwiki/get-status",
```
Comment (DavidePrincipi) [inline dokuwiki/imageroot/actions/get-status/validate-output.json:4]:
```suggestion
  "$id": "http://schema.nethserver.org/dokuwiki/get-status-output.json",
```

File: core/imageroot/usr/local/nethserver/agent/actions/get-status/validate-output.json [hunk 9]
```diff
@@ -0,0 +1,119 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Get module status",
+  "$id": ""http://schema.nethserver.org/agent/get-status-output.json",
```
Comment (andre8244) [inline core/imageroot/usr/local/nethserver/agent/actions/get-status/validate-output.json:4]:
```suggestion
  "$id": "http://schema.nethserver.org/agent/get-status-output.json",
```

**Review by andre8244 — CHANGES_REQUESTED** [review]
Just a small typo

File: core/imageroot/var/lib/nethserver/cluster/actions/get-module-info/validate-output.json [hunk 10]
```diff
@@ -0,0 +1,129 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "get-module-info output",
+  "description": "Get module info output",
+  "$id": "http://schema.nethserver.org/cluster/get-modules-info.json",
```
Comment (DavidePrincipi) [inline core/imageroot/var/lib/nethserver/cluster/actions/get-module-info/validate-output.json:5]:
```suggestion
  "$id": "http://schema.nethserver.org/cluster/get-module-info-output.json",
```

File: core/imageroot/var/lib/nethserver/cluster/actions/get-module-info/validate-input.json [hunk 11]
```diff
@@ -0,0 +1,21 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "info-module input",
+  "$id": "http://schema.nethserver.org/cluster/info-module-input.json",
```
Comment (DavidePrincipi) [inline core/imageroot/var/lib/nethserver/cluster/actions/get-module-info/validate-input.json:4]:
```suggestion
  "$id": "http://schema.nethserver.org/cluster/get-module-info-input.json",
```

**Review by DavidePrincipi — APPROVED** [review]

**Pull request merged**